arrow-select/src/take.rs

```diff
@@ -150,6 +150,11 @@ where
                 *length as u32,
             )?))
         }
+        DataType::Map(_, _) => {
+            let list_arr = ListArray::from(values.as_map());
+            let list_data = take_list::<_, Int32Type>(&list_arr, indices)?;
+            Ok(Arc::new(MapArray::try_from(&list_data)?))
```

Comment:
```suggestion
         let builder = list_data.into_builder().data_type(...);
            Ok(Arc::new(MapArray::from(builder.build_unchecked())?))
```
I.e. remove the try_from conversion?